Gem/Code/Source/RHI/XRExampleComponent.cpp

```diff
@@ -0,0 +1,377 @@
+
+            //Model matrix for the cube related to the front view
+            AZ::RPI::PoseData frontViewPoseData = xrSystem->GetViewFrontPose();
+            m_modelMatrices[0] = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(frontViewPoseData.orientation, frontViewPoseData.position) * initialScaleMat;
+                      
+            //Model matrix for the cube related to the left controller
+            AZ::RPI::PoseData controllerLeftPose = xrSystem->GetControllerPose(0);
+            AZ::Matrix4x4 leftScaleMat = initialScaleMat * AZ::Matrix4x4::CreateScale(AZ::Vector3(xrSystem->GetControllerScale(0), xrSystem->GetControllerScale(0), xrSystem->GetControllerScale(0)));
+            m_modelMatrices[1] = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(controllerLeftPose.orientation, controllerLeftPose.position) * leftScaleMat;
+
+            //Model matrix for the cube related to the right controller
+            AZ::Matrix4x4 rightScaleMat = initialScaleMat * AZ::Matrix4x4::CreateScale(AZ::Vector3(xrSystem->GetControllerScale(1), xrSystem->GetControllerScale(1), xrSystem->GetControllerScale(1)));
+            AZ::RPI::PoseData controllerRightPose = xrSystem->GetControllerPose(1);
+            m_modelMatrices[2] = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(controllerRightPose.orientation, controllerRightPose.position) * rightScaleMat;
+        }      
+        
+        for (int i = 0; i < s_numberOfCubes; ++i)
+        {
+            m_shaderResourceGroups[i]->SetConstant(m_shaderIndexWorldMat, m_modelMatrices[i]);
+            m_shaderResourceGroups[i]->SetConstant(m_shaderIndexViewProj, m_viewProjMatrix);
+            m_shaderResourceGroups[i]->Compile();
+        }
+
+        BasicRHIComponent::OnFramePrepare(frameGraphBuilder);
+    }
+
+    XRExampleComponent::SingleCubeBufferData XRExampleComponent::CreateSingleCubeBufferData([[maybe_unused]] const AZ::Vector4 color)
+    {
+        AZStd::vector<AZ::Vector4> vertexColor =
+        {
+            //Front Face
+            DarkBlue, DarkBlue, DarkBlue, DarkBlue,
+            //Back Face                                                                       
+            Blue, Blue, Blue, Blue,
+            //Left Face                                                                      
+            DarkGreen, DarkGreen, DarkGreen, DarkGreen,
+            //Right Face                                                                    
+            Green, Green, Green, Green,
+            //Top Face                                                                  
+            DarkRed, DarkRed, DarkRed, DarkRed,
+            //Bottom Face                                                                    
+            Red, Red, Red, Red,
+        };
+
+        // Create vertices, colors and normals for a cube and a plane
+        SingleCubeBufferData bufferData;
+        {
+            
+            AZStd::vector<AZ::Vector3> vertices =
```

Comment:
Same, `vertices` can be const and I'd try to use `fixed_vector` to use stack memory.